src/quick-lint-js/parse.h

```diff
@@ -2810,10 +2810,15 @@ class parser {
       this->skip();
       const char8 *end_of_else = this->lexer_.end_of_previous_token();
       bool has_left_paren = this->peek().type == token_type::left_paren;
-      parse_and_visit_body();
+      if (has_left_paren) {
+        this->parse_and_visit_expression(v);
+      } else {
+        parse_and_visit_body();
+      }
       bool has_left_curly = this->peek().type == token_type::left_curly;
       if (this->peek().has_leading_newline == false && has_left_paren &&
           has_left_curly) {
+        // if (cond) {} else (cond) {} // Invalid
         this->error_reporter_->report(error_missing_if_after_else{
             .expected_if = source_code_span(end_of_else, end_of_else),
         });
```

Comment:
Nit: I think this would be easier to understand if it was moved up to after line 2814. What do you think?

Reply:
I think it makes sense here.